lib/cache/senders/impressions_sender.rb

```diff
@@ -0,0 +1,59 @@
+module SplitIoClient
+  module Cache
+    module Senders
+      class ImpressionsSender
+        def initialize(impressions_repository, config, api_key)
+          @impressions_repository = impressions_repository
+          @config = config
+          @api_key = api_key
+        end
+
+        def call
+          # Disable impressions if @config.impressions_queue_size == -1
+          if @config.impressions_queue_size < 0
+            @config.logger.info('Disabling impressions service by config')
+            return
+          end
+
+          if ENV['SPLITCLIENT_ENV'] == 'test'
+            post_impressions
+          else
+            Thread.new do
+              @config.logger.info('Starting impressions service') unless ENV['SPLITCLIENT_ENV'] == 'test'
+
+              loop do
+                post_impressions
+
+                sleep(randomize_interval(@config.impressions_refresh_rate))
```

Comment:
do we need to cherry pick or rebase devleopment to make sure your Utilities class is used instead?